### PR TITLE
🐛 Remove `IdentityRef.Kind`

### LINKED ISSUE #1882: `identityRef.kind` is ignored
/kind bug

**What steps did you take and what happened:**

Identified while working on fix for #1881. The `identityRef` field has a `kind` child field to allow you to specify the kind of object we inspect. Alas, it is completely ignored. We should either remove this field, make it an enum (`["Secret"]`), or otherwise *not* ignore it.

**What did you expect to happen:**

The field should do something.

**Anything else you would like to add:**

None.

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built): `master`
- Cluster-API version: n/a
- OpenStack version: n/a
- Minikube/KIND version: n/a
- Kubernetes version (use `kubectl version`): n/a
- OS (e.g. from `/etc/os-release`): n/a

### PR DESCRIPTION
**Pull request by EmilienM**
**What this PR does / why we need it**:

Delete the `identityRef.Kind` field.
It was used to specify the kind of the identity provider to use but was actually ignored.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1882

**TODOs**:

- [X] squashed commits
- if necessary:
  - [X] includes documentation
  - [X] adds unit tests

### CODE REVIEW

**Review by mdbooth — APPROVED** [review]

File: api/v1alpha7/conversion.go [hunk 1]
```diff
@@ -678,3 +684,11 @@ func Convert_v1alpha7_SecurityGroup_To_v1alpha8_SecurityGroupStatus(in *Security
 
 	return nil
 }
+
+func Convert_v1alpha7_OpenStackIdentityReference_To_v1alpha8_OpenStackIdentityReference(in *OpenStackIdentityReference, out *infrav1.OpenStackIdentityReference, s apiconversion.Scope) error {
+	err := autoConvert_v1alpha7_OpenStackIdentityReference_To_v1alpha8_OpenStackIdentityReference(in, out, s)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (lentzi90) [inline api/v1alpha7/conversion.go:693]: nit
```suggestion
	return autoConvert_v1alpha7_OpenStackIdentityReference_To_v1alpha8_OpenStackIdentityReference(in, out, s)
```

**Review by lentzi90 — COMMENTED** [review]
/lgtm
Holding if you want to fix that nit
/hold

File: api/v1alpha7/conversion.go (hunk 1 above)
Comment (EmilienM) [reply api/v1alpha7/conversion.go]: done!

**Comment by EmilienM** [comment]
/hold cancel
I think it's ready.

**Comment by mdbooth** [comment]
Restoring @lentzi90's lgtm as there were no changes other than the requested nit fix.

/lgtm

**Pull request merged**